.github/workflows/publish-aws-bundle.yaml

```diff
@@ -0,0 +1,121 @@
+
+jobs:
+  publish:
+    name: Build and Publish AWS Bundle
+    runs-on: ubuntu-latest
+    # Security: Only run on authorized triggers
+    # - Tags (v*): Regular releases, anyone with push access can create tags
+    # - RC branches (rc-*): Release candidates, requires write access (includes PR merges)
+    # - workflow_dispatch: Manual trigger, requires write access (checked in step below)
+    permissions:
+      contents: read
+      id-token: write
+
+    steps:
+      - name: Verify authorization for manual trigger
+        if: github.event_name == 'workflow_dispatch'
+        run: |
+          # For manual triggers, verify the user has write access to the repository
+          # This prevents unauthorized users from triggering the workflow
+          # Note: GitHub's OIDC provider already validates repository_owner == 'Worklytics'
+          # but we add an extra check here for defense in depth
+          
+          # Check if actor is a member of the organization or has write access
+          # This is a basic check - GitHub will also enforce permissions
+          if [ "${{ github.event.sender.type }}" != "User" ] && [ "${{ github.event.sender.type }}" != "Bot" ]; then
+            echo "❌ ERROR: Unauthorized trigger. Only users and bots can trigger this workflow."
+            exit 1
+          fi
+          
+          echo "✓ Authorized trigger by: ${{ github.event.sender.login }}"
+          echo "  Actor type: ${{ github.event.sender.type }}"
+          echo "  Repository: ${{ github.repository }}"
+          echo ""
+          echo "Note: This workflow requires write access to the repository."
+          echo "GitHub will enforce this permission check before allowing the workflow to run."
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          java-version: '21'
+          distribution: 'temurin'
+          cache: 'maven'
+
+      - name: Configure AWS credentials via OIDC
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_ARN }}
+          aws-region: us-west-2
+
+      - name: Install dependencies
+        run: |
+          # Install jq if not already available
+          sudo apt-get update && sudo apt-get install -y jq
+
+      - name: Publish AWS Bundle
+        run: |
+          # Determine version from tag or pom.xml
+          REF_NAME="${{ github.ref_name }}"
+          
+          if [[ "$REF_NAME" =~ ^v[0-9]+\.[0-9]+\.[0-9]+ ]]; then
+            VERSION="${REF_NAME#v}"
+            echo "✓ Detected version tag: $REF_NAME, extracted version: $VERSION"
+          else
+            # Script will read from pom.xml
+            VERSION=""
+            echo "✓ Will read version from pom.xml"
+          fi
+          
+          # Export GH_REF for metadata
+          export GH_REF="$REF_NAME"
+          
+          # Call the publish script
```

Comment:
This seems more or less same as GCP case (extracting version from branch or POM: maybe we can extract this as a template and reuse it between two workflows